react/components/Table/Toolbar/index.js

```diff
@@ -0,0 +1,167 @@
+import React, { Fragment } from 'react'
+import PropTypes from 'prop-types'
+
+import InputToolbar from './InputToolbar'
+import ButtonDensity from './ButtonDensity'
+import ButtonFields from './ButtonFields'
+import ButtonDownload from './ButtonDownload'
+import ButtonUpload from './ButtonUpload'
+import ButtonNewLine from './ButtonNewLine'
+import ExtraActions from './ExtraActions'
+import ButtonToolbar from './ButtonToolbar'
+
+const Container = ({ justify = 'end', children }) => (
+  <div id="toolbar" className={`mb5 flex flex-row w-100 justify-${justify}`}>
+    {children}
+  </div>
+)
+
+Container.propTypes = {
+  justify: PropTypes.oneOf(['between', 'end', 'start', 'around', 'center']),
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
+}
+
+const Toolbar = ({
+  actions: {
+    inputSearch,
+    download,
+    upload,
+    fields,
+    extraActions,
+    newLine,
+    density,
+  },
+  loading,
+  children,
+}) => {
+  const isDownloadVisible = download && download.handleCallback
+  const isUploadVisible = upload && upload.handleCallback
+  const isFieldsVisible = fields && fields.showAllLabel && fields.hideAllLabel
+  const isExtraActionsVisible =
+    extraActions && extraActions.label && extraActions.actions.length > 0
+  const isNewLineVisible = newLine && newLine.label
+  const isSearchBarVisible = !!inputSearch
+  const isDensityVisible =
+    density &&
+    density.buttonLabel &&
+    density.lowOptionLabel &&
+    density.mediumOptionLabel &&
+    density.highOptionLabel
+
+  return (
+    children || (
+      <Container justify={isSearchBarVisible ? 'between' : 'end'}>
+        {inputSearch && (
+          <InputToolbar disabled={loading} inputSearch={inputSearch} />
+        )}
+        <div className="flex flex-row items-center">
+          <Fragment>
+            {isDensityVisible && (
```

Comment:
I don't really understand how this can be useful. And these buttons are exported as composites of the Toolbar, so when you receive a `children` you can organize them the way you want.

Reply:
How are they exported? I haven't seen this happening... I've seen that the table receives `download` and `upload` objects so... I really didn't get what you're saying about these composites.

The usage would be very simple

```jsx
function getButton(type, props) {
  switch (type) {
    case 'upload':
      return <Button icon={...}... />
   ...
  }
}

// usage
{isFieldsVisible && getButton('fields', {...})}
{isDownloadVisible && getButton('download', {...})}
{isUploadVisible && getButton('upload', {...})}
```

Or something like that

Reply:
I did it